src/olympia/addons/tests/test_serializers.py

```diff
@@ -470,6 +474,31 @@ def test_is_recommended(self):
         result = self.serialize()
         assert result['is_recommended'] is True
 
+    def test_promoted(self):
+        # With a promoted extension.
+        self.addon = addon_factory(recommended=True)
+
+        result = self.serialize()
+        promoted = result['promoted']
+        assert promoted['category'] == RECOMMENDED.api_name
+        assert promoted['apps'] == [app.short for app in amo.APP_USAGE]
+
+        # With a specific application.
+        self.addon.promotedaddon.update(application_id=amo.FIREFOX.id)
+        result = self.serialize()
+        assert result['promoted']['apps'] == [amo.FIREFOX.short]
+
+        # With a recommended theme.
+        self.addon = addon_factory(type=amo.ADDON_STATICTHEME)
+        featured_collection, _ = Collection.objects.get_or_create(
+            id=settings.COLLECTION_FEATURED_THEMES_ID)
+        featured_collection.add_addon(self.addon)
+
+        result = self.serialize()
+        promoted = result['promoted']
+        assert promoted['category'] == RECOMMENDED.api_name
+        assert promoted['apps'] == [app.short for app in amo.APP_USAGE]
```

Comment:
For some reason this test is failing, but only for the `ESAddonSerializer` and not for the plain `AddonSerializer`. It's failing because rather than finding all of the apps in `promoted['apps']`, it's just finding `['firefox']`, which seems to be leftover from the previous step in the test. I don't know why this is happening for just the `ESAddonSerializer` test.

Reply:
Ah, `TestESAddonSerializerOutput` assumes there's only one addon; the second `addon_factory` is creating another one , but the test just selects `[0]`, i.e the extension instead of the theme. (look for the "there should only be one addon in the index" comment.)

So, you can either change the test to modify the existing self.addon (change the addon type; delete the PromotedAddon) or split the theme test into a separate method.

Reply:
Great, thanks!